docs/cs_v6_release_guide.md

```diff
@@ -90,6 +90,8 @@ For `v7` we can then focus on other big features, like [SAIL](https://github.com
 
 - `clang-tidy` is now run on all files changed by a PR.
 - ASAN: All tests are now run with the address sanitizer enabled. This includes checking for leaks.
+- Many more asserts were added. They are only enabled for debug builds (with a few exceptions).
```

Comment:
I think we should add `It prints warnings but it does not abort the program` so that the user does not need to worry about abrupt SIGABRT.